### DISCUSSION TITLE
Welcome to ADE-Scheduler Discussions!

### DISCUSSION THREAD

**Discussion by jeertmans** [SnaKyEyeS/ADE-Scheduler#648] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vinraspa** [comment]
Hello,
I just found the ADE scheduler project.
I have been developing a similar project (ADEexplorer) for a few years but it is a local tool, not a web one as yours.
The frontend is rendered with pyQt5.My project is available here: https://gitlab.com/zigma12/ADEexplorer
I develop the tool from the teacher's point of view for the university of Clermont-Ferrand.
Congratulations for your very good work!

I have a question.
I work with ics files (as you do I guess) and I try to rebuild a coherence (knowing that the information of ics files is incomplete)
I know the existence of the webapi of ADE campus. I could test it on the university of Bourgogne (an acquaintance who gave me a login/pwd)
On the Clermont side, I don't have the necessary rights :` \
But I saw here or there that you said that login and password were not necessary to use ADE scheduler.
**Is there a way I don't know to make requests and get the famous xml file** without credentials?

Thanks in advance for your answers.
Vincent

**Reply by jeertmans** [reply]
Hello @vinraspa , it is always nice to meet people that also work on similar projects !

At first, we began with web scrapping, but we rapidly obtained credentials to get dedicated access to the ADE API.
Some people at UCLouvain, in charge of our IT services, created credentials for our web app that we can use to make requests.

Yes, we also work with ics files that we build upon the information found in the XML responses. XML parsing is done [here](https://github.com/ADE-Scheduler/ADE-Scheduler/blob/main/backend/ade_api.py) and events classes are defined [here](https://github.com/ADE-Scheduler/ADE-Scheduler/blob/main/backend/events.py)

In our case, we cannot do requests without credentials. with that credentials, we request every hour a token that allows us to get calendar information.
But I don't know about your university and what they allow or not. The best thing might be to contact the person in charge of maintaining your ADE service.

Our codebase became larger and larger over the time, so it might be hard to get an eye into it, so do not hesitate if you have specific questions about anything ;-)

Jérome

**Reply by vinraspa** [reply]
Thank you for your quick reply.
I hoped there was a secret way to access ADE database without credentials.
Now I understand your configuration is different from mine in that yours is a web app that interfaces between ADE and user. You only need one token.
I certainly will analyze closer your code because it looks pretty and efficient.
Thank you again!
Cheers
Vincent

**Reply by jeertmans** [reply]
Indeed, we do all our ADE requests under the same token. User logins are used for other purposes.
Thanks you for enthusiasm about our application!
Kind regards,
Jérome
